artemis-cli/src/test/java/org/apache/activemq/cli/test/CliProducerTest.java

```diff
@@ -88,6 +91,138 @@ public void testSendMessage() throws Exception {
+
+      m = consumeMessages(session, address, 1).get(0);
+      Assert.assertTrue(m.propertyExists("myShort"));
+      Assert.assertEquals(Short.MAX_VALUE, m.getShortProperty("myShort"));
+
+      assertEquals(0L, produceMessages(address, null, 1, ("[{'type':'short','key':'myShort','value':'" + Short.MAX_VALUE + 1 + "'}]").replaceAll("'", "\"")));
+   }
+
+   @Test
+   public void testFloatMessageProperties() throws Exception {
+      Message m;
+      String address = "test";
+      Session session = createSession(connection);
+
+      assertEquals(1L, produceMessages(address, null, 1, ("[{'type':'float','key':'myFloat','value':'" + Float.MAX_VALUE + "'}]").replaceAll("'", "\"")));
+
+      m = consumeMessages(session, address, 1).get(0);
+      Assert.assertTrue(m.propertyExists("myFloat"));
+      Assert.assertEquals(Float.MAX_VALUE, m.getFloatProperty("myFloat"), 0.0);
+
+      assertEquals(0L, produceMessages(address, null, 1, ("[{'type':'float','key':'myFloat','value':'badFloat'}]").replaceAll("'", "\"")));
+   }
+
+   @Test
+   public void testDoubleMessageProperties() throws Exception {
+      Message m;
+      String address = "test";
+      Session session = createSession(connection);
+
+      assertEquals(1L, produceMessages(address, null, 1, ("[{'type':'double','key':'myDouble','value':'" + Double.MAX_VALUE + "'}]").replaceAll("'", "\"")));
+
+      m = consumeMessages(session, address, 1).get(0);
+      Assert.assertTrue(m.propertyExists("myDouble"));
+      Assert.assertEquals(Double.MAX_VALUE, m.getDoubleProperty("myDouble"), 0.0);
+
+      assertEquals(0L, produceMessages(address, null, 1, ("[{'type':'double','key':'myDouble','value':'badDouble'}]").replaceAll("'", "\"")));
+   }
+
+   @Test
+   public void testStringMessageProperties() throws Exception {
+      Message m;
+      String address = "test";
+      Session session = createSession(connection);
+
+      assertEquals(1L, produceMessages(address, null, 1, "[{'type':'string','key':'myString','value':'foo'}]".replaceAll("'", "\"")));
+
+      m = consumeMessages(session, address, 1).get(0);
+      Assert.assertTrue(m.propertyExists("myString"));
+      Assert.assertEquals("foo", m.getStringProperty("myString"));
+   }
+
+   @Test
+   public void testBadMessageProperties() throws Exception {
+      PrintStream originalErr = System.err;
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      PrintStream newErr = new PrintStream(baos);
+      System.setErr(newErr);
+      try {
+         produceMessages("test", null, 1, "[{'type':'myType','key':'myKey','value':'myValue'}]".replaceAll("'", "\""));
+         assertEquals("Unable to set property: myKey. Did not recognize type: myType. Supported types are: boolean, int, long, byte, short, float, double, string.\n", baos.toString());
+      } finally {
+         System.setErr(originalErr);
+      }
```

Comment:
Ick on messing with the System. I believe most of the regular artemis CLI commands have a 'context' of some kind thats used for the output, which allows testing the output without this kind of hack. Doesnt the Producer?

Reply:
The `ProducerThread` logs directly using `System`. I sent #4846 to fix this. Once that's merged I'll update this test.